pyronear/models/ssresnet18.py

```diff
@@ -0,0 +1,78 @@
+from torchvision.models.resnet import ResNet, BasicBlock
+import torchvision
+from torch import nn
+import torch
+import torch.nn.functional as F
+
+
+class SSResNet18(ResNet):
+    def __init__(self, frame_per_seq):
+        super(SSResNet18, self).__init__(BasicBlock, [2, 2, 2, 2])
+        dictres18 = torchvision.models.resnet18(pretrained=True).state_dict()
+        self.load_state_dict(dictres18)
+
+        self.frame_per_seq = frame_per_seq
+        self.fc = nn.Linear(256, 1)
+        self.conv1_1 = nn.Conv2d(512 * self.frame_per_seq, 512, kernel_size=1)
+        self.conv3_1 = nn.Conv2d(512, 512, kernel_size=3)
+        self.conv1_2 = nn.Conv2d(512, 256, kernel_size=1)
+        self.conv2_bn = nn.BatchNorm2d(512)
+        self.conv2_bn1 = nn.BatchNorm2d(512)
+        self.conv2_bn2 = nn.BatchNorm2d(256)
+        self.sig = nn.Sigmoid()
+
+        nn.init.kaiming_normal_(self.conv1_1.weight, mode='fan_out', nonlinearity='relu')
+        nn.init.kaiming_normal_(self.conv3_1.weight, mode='fan_out', nonlinearity='relu')
+        nn.init.kaiming_normal_(self.conv1_2.weight, mode='fan_out', nonlinearity='relu')
+
+        nn.init.constant_(self.conv2_bn.weight, 1)
+        nn.init.constant_(self.conv2_bn.bias, 0)
+
+        nn.init.constant_(self.conv2_bn1.weight, 1)
+        nn.init.constant_(self.conv2_bn1.bias, 0)
+
+        nn.init.constant_(self.conv2_bn2.weight, 1)
+        nn.init.constant_(self.conv2_bn2.bias, 0)
```

Comment:
Each layer type shares the same initialization, try to consider doing something like this instead: https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py#L160

Since you would be moving the pretrained option out of the class to the function, this would be a better option ;)